logging/slog_logger.go

```diff
@@ -0,0 +1,75 @@
+package logging
+
+import (
+	"fmt"
+	"log/slog"
+	"os"
+)
+
+type SLogger struct {
+	logger *slog.Logger
+}
+
+var _ Logger = (*SLogger)(nil)
+
+func NewSlogLogger(env LogLevel) Logger {
+	if env == Production {
+		logHandler := slog.NewJSONHandler(os.Stdout, nil).WithAttrs([]slog.Attr{slog.String("env", string(env))})
+		logger := slog.New(logHandler)
+		return &SLogger{
+			logger: logger,
+		}
+	} else if env == Development {
+		// Even though code is same for both environments, we are keeping it separate to show that we can have different
+		// implementations for different environments. For example, we can have log levels set differently
+		logHandler := slog.NewJSONHandler(os.Stdout, nil).WithAttrs([]slog.Attr{slog.String("env", string(env))})
```

Comment:
why is it the same? is there a way to not show debug logs in production mode?

Reply:
yea - only `env` is different. since `slog` doesn't have different constructor. and basically dev and prod will both need structured log. ideally local will need more terminal like

Reply:
not following but it's fine with me since I won't be using slog for the time being, so someone else should request changes here if this is not fine for them, or we can change this later.

Reply:
oh wait. I read your comment wrong 🤦 . I think we can set the log level differently in prod, like only warn and above.

Reply:
actually by default it is `Info` only, to enable `debug` you have to explicitly set it to debug.